src/core/components/button/LinkButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { ReactElement, ReactNode, AnchorHTMLAttributes } from 'react';
+import { SerializedStyles } from '@emotion/react';
+import { buttonStyles, buttonContents } from './shared';
+import type {
+	SharedButtonProps,
+	ButtonPriority,
+	Size,
+	IconSide,
+} from './types';
+
+export interface LinkButtonProps
+	extends SharedButtonProps,
+		AnchorHTMLAttributes<HTMLAnchorElement> {
+	priority?: ButtonPriority;
+	size?: Size;
+	iconSide?: IconSide;
+	icon?: ReactElement;
+	nudgeIcon?: boolean;
+	hideLabel?: boolean;
+	children: ReactNode;
+	cssOverrides?: SerializedStyles | SerializedStyles[];
```

Comment:
Thanks @jamie-lynch. I've removed all the overridden props and it still passes type check. I guess we don't even need `children` because it's inherited from `AnchorHTMLAttributes<HTMLAnchorElement>` (and ultimately `DOMAttributes<T>`)?